cylc/flow/network/client.py

```diff
@@ -223,15 +224,28 @@ def serial_request(self, command, args=None, timeout=None):
         self.loop.run_until_complete(task)
         return task.result()
 
-    @staticmethod
-    def get_header() -> dict:
+    def get_header(self) -> dict:
         """Return "header" data to attach to each request for traceability.
 
         Returns:
             dict: dictionary with the header information, such as
                 program and hostname.
         """
-        cmd = sys.argv[0]
+
+        ssh_connection = os.getenv("SSH_CONNECTION")
+        host = socket.gethostname()
+        if ssh_connection is None:
+            comms_method = 'local command'
+        elif self.host and (
+                socket.gethostbyname(host) ==
+                socket.gethostbyname(str(self.host))):
+            comms_method = CommsMeth.SSH
```

Comment:
Could this be `get_comms_method`?

Many users SSH into their systems to gain access so `SSH_CONNECTION` is non-null even for "local" usage.

Reply:
I've run some tests and `get_comms_method` wont work here unfortunately, since for configured SSH comms at this stage we are on the scheduler which will return zmq for the comms method. I can see my workaround of looking at the latest SSH command and comparing hosts would be problematic in the case where users SSH first. I shall have a little think about this and run some tests to see if I can get a more reliable way of returning the correct communication method.

Reply:
Can we hardcode this on the client:

```python
# cylc.flow.network.client
class SuiteRuntimeClient:
    COMMS_METH = 'zmq'

    def get_headers(cls):
    ...
    return {
        'meta': {
            ...
            'comms method': cls.COMMS_METH
        }
    }

# cylc.flow.network.ssh_client:
    COMMS_METH = 'ssh'
```

Reply:
Update, COMMS_METH could not be hard coded (the zmq would overwrite the ssh one) and had to be added as an environment variable (CLIENT_COMMS_METH) and sent via the ssh calling `cylc client`.